src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -1561,6 +1561,53 @@ LogicalResult ONNXUnsqueezeOp::inferShapes() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+
+// Squeeze
+
+LogicalResult ONNXSqueezeOp::inferShapes() {
+  if (!data().getType().isa<RankedTensorType>())
+    return emitError("Input tensor not ranked");
+
+  auto operandTy = data().getType().cast<RankedTensorType>();
+  int64_t inRank = operandTy.getRank();
+
+  ArrayAttr axisAttrs = axesAttr();
+  SmallVector<int64_t, 4> axes;
+  if (axisAttrs) {
```

Comment:
Coding style, to check error, much better to have this:

if (error) return
// code without error

instead of this long "then"